From 6a9b0728c55b8ad5c6d5dba3f38697eded09538a Mon Sep 17 00:00:00 2001
From: Lee Duncan <lduncan@suse.com>
Date: Wed, 4 Mar 2020 12:59:10 -0800
Subject: [PATCH 151/170] Fix iscsi.service so it handles restarts better

Requiring iscsid.service means that a restart of iscsi.service
restarted iscsid.service when unneccesary.

Also, we should treat an exit value of 15 as normal, since
this just means the session is already present.
It should rely on iscsid.socket, no iscsid.service.
---
 etc/systemd/iscsi.service | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/etc/systemd/iscsi.service b/etc/systemd/iscsi.service
index e475888..1c286d1 100644
--- a/etc/systemd/iscsi.service
+++ b/etc/systemd/iscsi.service
@@ -3,7 +3,7 @@ Description=Login and scanning of iSCSI devices
 Documentation=man:iscsiadm(8) man:iscsid(8)
 Before=remote-fs.target
 After=network.target network-online.target iscsid.service
-Requires=iscsid.service
+Requires=iscsid.socket
 ConditionPathExists=/etc/iscsi/initiatorname.iscsi
 
 [Service]
@@ -11,7 +11,7 @@ Type=oneshot
 ExecStart=/sbin/iscsiadm -m node --loginall=automatic
 ExecStop=/sbin/iscsiadm -m node --logoutall=automatic
 ExecStop=/sbin/iscsiadm -m node --logoutall=manual
-SuccessExitStatus=21
+SuccessExitStatus=21 15
 RemainAfterExit=true
 
 [Install]
-- 
2.21.1 (Apple Git-122.3)

